### PR TITLE
[Monitor Query] Add samples/docs for running big queries and overcoming API limits for size and record count

### LINKED ISSUE #30986: [Monitor Query] Add samples/docs for running big queries and overcoming API limits for size and record count
For details of the ask, see https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1597918.

### PR DESCRIPTION
**Pull request by jairmyree**
Closes #30986

### CODE REVIEW

**Comment by jairmyree** [comment]
Need to update the README.md to link to the new documentation

File: sdk/monitor/azure-monitor-query/LargeLogQuery.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# Running Large Log Queries using Log Analytics
+
+This guide will show you how to run large log queries **without** extra costs using Log Analytics and the Monitor Query SDK.
+
+Due to Log Analytics [service limits](https://learn.microsoft.com/en-us/azure/azure-monitor/service-limits#la-query-api), sometimes it may
+not be possible to retrieve all the expected data in a single query. For example, the nnumber of rows returned or the maximum size of the
```
Comment (lmolkova) [inline sdk/monitor/azure-monitor-query/LargeLogQuery.md:6]:
```suggestion
not be possible to retrieve all the expected data in a single query. For example, the number of rows returned or the maximum size of the
```

File: sdk/monitor/azure-monitor-query/LargeLogQuery.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+Set the `LOG_WORKSPACE_ID` variable to the ID of the Log Analytics workspace you want to query.
+
+```java com.azure.monitor.query.LargeQuerySample-setWorkspaceId
+static final String WORKSPACE_ID = "{workspace-id}";
+```
+
+### Defining the Helper Function
+
+In order to overcome the service limits, the strategy is to query data in smaller chunks based on some time column (i.e. TImeGenerated). The
```
Comment (lmolkova) [inline sdk/monitor/azure-monitor-query/LargeLogQuery.md:35]:
```suggestion
In order to overcome the service limits, the strategy is to query data in smaller chunks based on a timestamp column (e.g. `TimeGenerated`). The
```

File: sdk/monitor/azure-monitor-query/LargeLogQuery.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        }
+        else {
+            queryString = String.format("%1$s | where %4$s >= datetime('%2$s') and %4$s < datetime('%3$s')", originalQuery, endpoints.get(i), endpoints.get(i + 1), timeColumn);
+        }
+        batchQuery.addWorkspaceQuery(WORKSPACE_ID,queryString, null);
+    }
+
+    return batchQuery;
+}
+```
```
Comment (lmolkova) [inline sdk/monitor/azure-monitor-query/LargeLogQuery.md:57]: mind adding a comment showing an example of the original query and individual pieces of batch query after splitting? (either in the code or just in the md file)

File: sdk/monitor/azure-monitor-query/LargeLogQuery.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    for (int i = 0; i < endpoints.size(); i++) {
+        String queryString;
+        if (i == endpoints.size() - 1) {
+            queryString = String.format("%1$s | where %3$s >= datetime('%2$s')", originalQuery, endpoints.get(i), timeColumn);
+        }
+        else {
+            queryString = String.format("%1$s | where %4$s >= datetime('%2$s') and %4$s < datetime('%3$s')", originalQuery, endpoints.get(i), endpoints.get(i + 1), timeColumn);
+        }
+        batchQuery.addWorkspaceQuery(WORKSPACE_ID,queryString, null);
```
Comment (lmolkova) [inline sdk/monitor/azure-monitor-query/LargeLogQuery.md:52]:
```suggestion
        batchQuery.addWorkspaceQuery(WORKSPACE_ID, queryString, null);
```

File: sdk/monitor/azure-monitor-query/LargeLogQuery.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+}
+```
+
+### Defining the Query-BatchQuery conversion Functions
+
+This sample provids two functions to convert a `LogsQuery` into a `LogsBatchQuery`. One function splits the query into multiple smaller queries
+based upon the approximate* maximum amount of rows to be present in each query. The other function splits the query into multiple smaller queries based upon
+the approximate* maximum size of the data in bytes to be present in each query.
+
+_*If multiple rows have the exact same TimeGenerated, the queries in the batch query may exceed the maximum amount of rows/bytes_
```
Comment (lmolkova) [inline sdk/monitor/azure-monitor-query/LargeLogQuery.md:65]:
```suggestion
_*If multiple rows have the exact same `TimeGenerated` (or another timestamp column), the queries in the batch query may exceed the maximum amount of rows/bytes_
```

File: sdk/monitor/azure-monitor-query/LargeLogQuery.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+#### Splitting by Rows
+
+```java com.azure.monitor.query.LargeQuerySample-createBatchQueryFromLargeQueryByRows
+static LogsBatchQuery createBatchQueryFromLargeQueryByRowCount(String originalQuery,
+                                                               int maxRowPerBatch,
+                                                               String timeColumn) {
+
+    String findBatchEndpointsQuery = String.format(
```
Comment (lmolkova) [inline sdk/monitor/azure-monitor-query/LargeLogQuery.md:74]: same here, would you mind adding an example with the original query and an example of formatted string?

File: sdk/monitor/azure-monitor-query/LargeLogQuery.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+}
+```
+
+#### Splitting by Size
+
+```java com.azure.monitor.query.LargeQuerySample-createBatchQueryFromLargeQueryBySize
+static LogsBatchQuery createBatchQueryFromLargeQueryByByteSize(String originalQuery,
+                                                               int maxByteSizePerBatch,
+                                                               String timeColumn) {
+    String findBatchEndpointsQuery = String.format(
```
Comment (lmolkova) [inline sdk/monitor/azure-monitor-query/LargeLogQuery.md:102]: and here :)

File: sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/LargeQuerySample.java [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+        for (int i = 0; i < endpoints.size(); i++) {
+            String queryString;
+            if (i == endpoints.size() - 1) {
+                queryString = String.format("%1$s | where %3$s >= datetime('%2$s')", originalQuery, endpoints.get(i), timeColumn);
+            }
+            else {
+                queryString = String.format("%1$s | where %4$s >= datetime('%2$s') and %4$s < datetime('%3$s')", originalQuery, endpoints.get(i), endpoints.get(i + 1), timeColumn);
+            }
+            batchQuery.addWorkspaceQuery(WORKSPACE_ID,queryString, null);
```
Comment (lmolkova) [inline sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/LargeQuerySample.java:51]: nit 
```suggestion
            batchQuery.addWorkspaceQuery(WORKSPACE_ID, queryString, null);
```

File: sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/codesnippets/LogsLargeQueryMarkdownCodeSnippets.java [hunk 9]
```diff
@@ -0,0 +1,7 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (lmolkova) [inline sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/codesnippets/LogsLargeQueryMarkdownCodeSnippets.java:1]: is this file necessary?

File: sdk/monitor/azure-monitor-query/README.md [hunk 10]
```diff
@@ -405,6 +405,9 @@ raw JSON response. For example:
 }
 ```
 
+#### Overcoming Log Analytics query size limitations
+If your query exceeds the service limits, you can refer to the large log query documentation to learn how to overcome them.
```
Comment (lmolkova) [inline sdk/monitor/azure-monitor-query/README.md:409]: it's likely hard to add a link right away without failing the pipeline (only possible to add a permalink to your fork), but as a follow-up PR, mind adding a link to the other doc?

**Review by lmolkova — APPROVED** [review]
LGTM overall! I just left several suggestions to provide examples of formatted queries and a few minor comments.

File: sdk/monitor/azure-monitor-query/LargeLogQuery.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+```java com.azure.monitor.query.LargeQuerySample-createBatchQueryFromLargeQuery
+static LogsBatchQuery createBatchQueryFromLargeQuery(String originalQuery,
+                                                     List<OffsetDateTime> endpoints,
+                                                     String timeColumn) {
+    LogsBatchQuery batchQuery = new LogsBatchQuery();
+
+    for (int i = 0; i < endpoints.size(); i++) {
+        String queryString;
+        if (i == endpoints.size() - 1) {
+            queryString = String.format("%1$s | where %3$s >= datetime('%2$s')", originalQuery, endpoints.get(i), timeColumn);
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/LargeLogQuery.md:47]: We should not modify the query string. We should instead use the QueryTimeInterval to set the start and end timestamps.

File: sdk/monitor/azure-monitor-query/LargeLogQuery.md [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+
+### Defining the Helper Function
+
+In order to overcome the service limits, the strategy is to query data in smaller chunks based on a timestamp column (e.g. `TimeGenerated`). The
+following helper function take a large query and splits it into a `LogsBatchQuery` based on a list of time ranges (called endpoints).
+
+```java com.azure.monitor.query.LargeQuerySample-createBatchQueryFromLargeQuery
+static LogsBatchQuery createBatchQueryFromLargeQuery(String originalQuery,
+                                                     List<OffsetDateTime> endpoints,
+                                                     String timeColumn) {
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/LargeLogQuery.md:41]: How does the user get the list of endpoints? Also, endpoints can be confused for URLs. So, we should use something like dateRanges.

File: sdk/monitor/azure-monitor-query/LargeLogQuery.md [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+
+    for (int i = 0; i < endpoints.size(); i++) {
+        String queryString;
+        if (i == endpoints.size() - 1) {
+            queryString = String.format("%1$s | where %3$s >= datetime('%2$s')", originalQuery, endpoints.get(i), timeColumn);
+        }
+        else {
+            queryString = String.format("%1$s | where %4$s >= datetime('%2$s') and %4$s < datetime('%3$s')", originalQuery, endpoints.get(i), endpoints.get(i + 1), timeColumn);
+        }
+        batchQuery.addWorkspaceQuery(WORKSPACE_ID,queryString, null);
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/LargeLogQuery.md:52]: Instead of `null`, we should use the QueryTimeInterval type to set the start and end times.

File: sdk/monitor/azure-monitor-query/LargeLogQuery.md [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    return batchQuery;
+}
+```
+
+### Defining the Query-BatchQuery conversion Functions
+
+This sample provids two functions to convert a `LogsQuery` into a `LogsBatchQuery`. One function splits the query into multiple smaller queries
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/LargeLogQuery.md:61]:
```suggestion
This sample provides two functions to convert a `LogsQuery` into a `LogsBatchQuery`. One function splits the query into multiple smaller queries
```

File: sdk/monitor/azure-monitor-query/LargeLogQuery.md [hunk 15]
```diff
@@ -0,0 +1,123 @@
+# Running Large Log Queries using Log Analytics
+
+This guide will show you how to run large log queries **without** extra costs using Log Analytics and the Monitor Query SDK.
+
+Due to Log Analytics [service limits](https://learn.microsoft.com/en-us/azure/azure-monitor/service-limits#la-query-api), sometimes it may
+not be possible to retrieve all the expected data in a single query. For example, the number of rows returned or the maximum size of the
+data returned may exceed the stated limits. One approach for overcoming these limits is to split the queries into multiple smaller queries
+using different time ranges.
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/LargeLogQuery.md:8]: Queries can be split into smaller sub-queries based on time ranges only for certain types of queries - for e.g. aggregations may not work well.

File: sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/LargeQuerySample.java [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.util.Configuration;
+import com.azure.identity.ClientSecretCredential;
+import com.azure.identity.ClientSecretCredentialBuilder;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.monitor.query.models.*;
+
+public class LargeQuerySample {
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/LargeQuerySample.java:18]: This should be a runnable sample that works end to end. Also, add javadocs for the sample.

File: sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/LargeQuerySample.java [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+
+
+    // BEGIN: com.azure.monitor.query.LargeQuerySample-setWorkspaceId
+    static final String WORKSPACE_ID = "{workspace-id}";
+    // END: com.azure.monitor.query.LargeQuerySample-setWorkspaceId
+
+
+    static LogsQueryClient client;
+
+    public static void main(String[] args) {
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/LargeQuerySample.java:28]: Main method should demonstrate starting with a query that gets split into multiple sub-queries and show how to consolidate the results of sub-queries into a single result of the original query.

File: sdk/monitor/azure-monitor-query/src/test/java/com/azure/monitor/query/LogsQueryClientTest.java [hunk 18]
```diff
@@ -23,6 +23,10 @@
 import com.azure.monitor.query.models.LogsQueryOptions;
 import com.azure.monitor.query.models.LogsQueryResult;
 import com.azure.monitor.query.models.LogsQueryResultStatus;
+import com.azure.monitor.query.models.LogsTable;
+import com.azure.monitor.query.models.LogsTableCell;
+import com.azure.monitor.query.models.LogsTableColumn;
+import com.azure.monitor.query.models.LogsTableRow;
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/test/java/com/azure/monitor/query/LogsQueryClientTest.java:29]: These unused imports should be removed.

File: sdk/monitor/azure-monitor-query/README.md (hunk 10 above)
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/README.md:409]: As a rule of thumb, avoid gerunds in headings. Also, this "large log query documentation" text should be a link.
```suggestion
#### Overcome Log Analytics query size limitations

If your query exceeds the service limits, see the large log query documentation to learn how to overcome them.
```

File: sdk/monitor/azure-monitor-query/src/samples/java/README.md [hunk 19]
```diff
@@ -35,6 +35,36 @@ library.
 
 Troubleshooting steps can be found [here][SDK_README_TROUBLESHOOTING].
 
+## Running Large Log Queries using Log Analytics
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/src/samples/java/README.md:38]:
```suggestion
## Run large log queries using Log Analytics
```

File: sdk/monitor/azure-monitor-query/src/samples/java/README.md [hunk 20]
```diff
@@ -35,6 +35,36 @@ library.
...
+
+Due to Log Analytics [service limits][monitor_service_limits], sometimes it may
+not be possible to retrieve all the expected data in a single query. For example, the number of rows returned or the maximum size of the
+data returned may exceed the stated limits. One approach for overcoming these limits is to split the queries into multiple smaller queries
+using different time ranges.
+
+This work-around allows you to avoid the cost of exporting data to a storage account (and potentially the cost of the storage account as well).
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/src/samples/java/README.md:45]:
```suggestion
This workaround allows you to avoid the cost of exporting data to a storage account (and potentially the cost of the storage account as well). 
```

File: sdk/monitor/azure-monitor-query/src/samples/java/README.md [hunk 21]
```diff
@@ -35,6 +35,36 @@ library.
...
+Due to Log Analytics [service limits][monitor_service_limits], sometimes it may
+not be possible to retrieve all the expected data in a single query. For example, the number of rows returned or the maximum size of the
+data returned may exceed the stated limits. One approach for overcoming these limits is to split the queries into multiple smaller queries
+using different time ranges.
+
+This work-around allows you to avoid the cost of exporting data to a storage account (and potentially the cost of the storage account as well). 
+
+**Disclaimer:** This approach of splitting data retrieval into smaller queries is good when dealing with a few GBs of data or a few millions
+of records per hour. For larger data sets,
+[exporting][logs_data_export] is recommended.
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/src/samples/java/README.md:49]:
```suggestion
**Disclaimer:** This approach of splitting data retrieval into smaller queries is useful when dealing with a few GBs of data or a few millions records per hour. For larger data sets, [exporting][logs_data_export] is recommended.
```

File: sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/OvercomingLargeQueryLimitationsSample.java [hunk 22]
```diff
@@ -0,0 +1,171 @@
...
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.monitor.query.models.LogsBatchQuery;
+import com.azure.monitor.query.models.LogsBatchQueryResult;
+import com.azure.monitor.query.models.LogsQueryResult;
+import com.azure.monitor.query.models.LogsTable;
+import com.azure.monitor.query.models.LogsTableRow;
+import com.azure.monitor.query.models.QueryTimeInterval;
+
+/**
+ * A sample to demonstrate using a batch query to overcome the service limits for a large query.
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/OvercomingLargeQueryLimitationsSample.java:20]:
```suggestion
 * A sample to demonstrate using a batch logs query to overcome the service limits for a large query.
```

**Review by scottaddie — COMMENTED** [review]
In the samples README, please also update this text: 
image
Notice it mentions a completely different library.

File: sdk/monitor/azure-monitor-query/src/samples/java/README.md [hunk 23]
```diff
@@ -35,6 +35,36 @@ library.
...
+
+This sample shows how to parition a large query into smaller queries using the `LogsBatchQuery` class. The sample provides examples for 
+row-based partitioning and byte-size partitioning. The partitioning is based on the timestamp "TimeGenerated". 
+
+This sample is suitable for simple data retrieval queries that utilize a subset of KQL known as "Reduced KQL". It contains the following
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/src/samples/java/README.md:54]: Is there a doc that lists the operators? If so, link to that instead of listing them here.

File: sdk/monitor/azure-monitor-query/README.md (hunk 10 above)
Comment (jairmyree) [reply sdk/monitor/azure-monitor-query/README.md:409]: @scottaddie I plan on linking via a GitHub permalink once the PR merges.

File: sdk/monitor/azure-monitor-query/src/samples/java/README.md (hunk 23 above)
Comment (scottaddie) [reply sdk/monitor/azure-monitor-query/src/samples/java/README.md:52]: In that case, it's best to not invent the term here. Let's remove "known as Reduced KQL" from the sentence.

Comment (pvaneck) [reply sdk/monitor/azure-monitor-query/src/samples/java/README.md:52]: Here's the link I used which references this subset of operators: https://learn.microsoft.com/azure/azure-monitor/logs/basic-logs-query?tabs=portal-1#kql-language-limits

File: sdk/monitor/azure-monitor-query/src/samples/java/README.md [hunk 24]
```diff
@@ -35,6 +35,34 @@ library.
...
+## Run large log queries using Log Analytics
+
+Due to Log Analytics [service limits][monitor_service_limits], sometimes it may
+not be possible to retrieve all the expected data in a single query. For example, the number of rows returned or the maximum size of the
+data returned may exceed the stated limits. One approach for overcoming these limits is to split the queries into multiple smaller queries
+using different time ranges.
+
+This workaround allows you to avoid the cost of exporting data to a storage account (and potentially the cost of the storage account as well). 
+
+**Disclaimer:** This approach of splitting data retrieval into smaller queries is useful when dealing with a few GBs of data or a few millions records per hour. For larger data sets, [exporting][logs_data_export] is recommended.
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/src/samples/java/README.md:47]:
```suggestion
**Disclaimer:** This approach of splitting data retrieval into smaller queries is useful when dealing with a few GBs of data or a few million records per hour. For larger data sets, [exporting][logs_data_export] is recommended.
```

File: sdk/monitor/azure-monitor-query/src/samples/java/README.md [hunk 25]
```diff
@@ -35,6 +35,34 @@ library.
...
+
+This sample shows how to parition a large query into smaller queries using the `LogsBatchQuery` class. The sample provides examples for
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/src/samples/java/README.md:49]:
```suggestion
This sample shows how to partition a large query into smaller queries using the `LogsBatchQuery` class. The sample provides examples for 
```

File: sdk/monitor/azure-monitor-query/src/samples/java/README.md [hunk 26]
```diff
@@ -23,8 +23,8 @@ Getting started explained in detail [here][SDK_README_GETTING_STARTED].
 
 ## Examples
 
-The following sections provide code samples covering common operations with OpenTelemetry Azure Monitor Exporter client
-library.
+The following sections provide code samples covering common operations with Azure Monitor Logs and Metrics query client
+libraries.
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/src/samples/java/README.md:27]:
```suggestion
The following sections provide code samples covering common operations with the Azure Monitor Query client library.
```

File: sdk/monitor/azure-monitor-query/README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (jairmyree) [reply sdk/monitor/azure-monitor-query/README.md]: @srnagar I can't link to the documentation I added until the PR gets merged because it'll fail the ci. I'm going to have a follow-up linking to the documentation that is in the samples README file. I'll update this text to reflect that.

File: sdk/monitor/azure-monitor-query/src/samples/java/README.md [hunk 27]
```diff
@@ -35,6 +34,23 @@ library.
...
+
+This workaround allows you to avoid the cost of exporting data to a storage account (and potentially the cost of the storage account as well). 
+
+**Disclaimer:** This approach of splitting data retrieval into smaller queries is useful when dealing with a few GBs of data or a few million records per hour. For larger data sets, [exporting][logs_data_export] is recommended.
+
+This sample shows how to partition a large query into smaller queries using the `LogsBatchQuery` class. The sample provides examples for 
+row-based partitioning and byte-size partitioning. The partitioning is based on the timestamp "TimeGenerated". 
+
+This sample is suitable for simple data retrieval queries that utilize a subset of KQL operators. The subset of KQL operators supported are can
+be found [here][kql_language_subset].
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/src/samples/java/README.md:52]:
```suggestion
This sample is suitable for simple data retrieval queries that utilize a subset of KQL operators. The subset of supported KQL operators can be found [here][kql_language_subset].
```

File: sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/OvercomingLargeQueryLimitationsSample.java [hunk 28]
```diff
@@ -0,0 +1,171 @@
...
+        }
+
+        return queryTimeIntervals;
+    }
+
+    /**
+     * Helper method to create a list of time intervals for a batch query based on the row count limit.
+     *
+     * @param originalQuery The original query string.
+     * @param maxRowPerBatch The maximum row count per batch. If multiple log entries returned in the original query
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/OvercomingLargeQueryLimitationsSample.java:123]:
```suggestion
     * @param maxRowsPerBatch The maximum row count per batch. If multiple log entries returned in the original query
```

File: sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/OvercomingLargeQueryLimitationsSample.java [hunk 29]
```diff
@@ -0,0 +1,171 @@
...
+     *                       have the exact same timestamp, the row count per batch may exceed this limit.
+     * @return A list of {@link QueryTimeInterval} objects.
+     */
+    static List<QueryTimeInterval> createQueryTimeIntervalsForBatchQueryByRowCount(String originalQuery,
+                                                                   int maxRowsPerBatch) {
+
+        String findBatchEndpointsQuery = String.format(
+            "%1$s | sort by TimeGenerated desc | extend batch_num = row_cumsum(1) / %2$s | summarize batchStart=min(TimeGenerated) by batch_num | sort by batch_num desc | project batchStart",
+            originalQuery,
+            maxRowPerBatch);
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/OvercomingLargeQueryLimitationsSample.java:133]:
```suggestion
            maxRowsPerBatch);
```

File: sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/SplitQueryByRowSample.java [hunk 30]
```diff
@@ -0,0 +1,113 @@
...
+            queryTimeIntervals.add(timeInterval);
+        }
+
+        return queryTimeIntervals;
+    }
+
+    static LogsQueryResult simulateSingleQuery(List<LogsBatchQueryResult> batchQueryResults) {
+        List<LogsTable> logsTables = new ArrayList<>();
+        for (LogsBatchQueryResult batchQueryResult : batchQueryResults) {
+            logsTables.addAll(batchQueryResult.getAllTables());
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/SplitQueryByRowSample.java:109]: Same here - the result should not contain multiple tables. Instead, it should contain a single table with all rows from every sub-request in the batch.

File: sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/SplitQueryByRowSample.java [hunk 31]
```diff
@@ -0,0 +1,113 @@
...
+            OffsetDateTime startTime = offsetDateTimes.get(i);
+            OffsetDateTime endTime = i == offsetDateTimes.size() - 1 ? OffsetDateTime.now() : offsetDateTimes.get(i + 1);
+            QueryTimeInterval timeInterval = new QueryTimeInterval(startTime, endTime);
+            queryTimeIntervals.add(timeInterval);
+        }
+
+        return queryTimeIntervals;
+    }
+
+    static LogsQueryResult simulateSingleQuery(List<LogsBatchQueryResult> batchQueryResults) {
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/SplitQueryByRowSample.java:106]: Update method name to `combineResults`

File: sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/SplitQueryByRowSample.java [hunk 32]
```diff
@@ -0,0 +1,113 @@
...
+import com.azure.monitor.query.models.LogsQueryResult;
+import com.azure.monitor.query.models.LogsTable;
+import com.azure.monitor.query.models.LogsTableRow;
+import com.azure.monitor.query.models.QueryTimeInterval;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SplitQueryByRowSample {
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/SplitQueryByRowSample.java:19]:
```suggestion
public class SplitQueryByRowCountSample {
```

File: sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/SplitQueryByByteSample.java [hunk 33]
```diff
@@ -0,0 +1,126 @@
...
+import com.azure.monitor.query.models.LogsBatchQueryResult;
+import com.azure.monitor.query.models.LogsQueryResult;
+import com.azure.monitor.query.models.LogsTable;
+import com.azure.monitor.query.models.LogsTableRow;
+import com.azure.monitor.query.models.QueryTimeInterval;
+
+/**
+ * A sample to demonstrate using a batch logs query to overcome the service limits for a large query.
+ */
+public class SplitQueryByByteSample {
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/SplitQueryByByteSample.java:22]:
```suggestion
public class SplitQueryByByteSizeSample {
```

File: sdk/monitor/azure-monitor-query/README.md [hunk 34]
```diff
@@ -405,6 +405,10 @@ raw JSON response. For example:
 }
 ```
 
+#### Overcome Log Analytics query size limitations
+
+If your query exceeds the [service limits][service_limits], see the large log query documentation to learn how to overcome them.
```
[the remark this answers is not in the document]
Comment (jairmyree) [reply sdk/monitor/azure-monitor-query/README.md:410]: I'll add this after this PR merges.

**Review by srnagar — APPROVED** [review]
LGTM!
